elki-core-math/src/main/java/elki/math/statistics/dependence/mcde/MWPTest.java

```diff
@@ -100,14 +106,14 @@ private MWPTest() {
 
   @Override
   public <A> MWPRanking correctedRanks(NumberArrayAdapter<?, A> adapter, A data, int len) {
-    int[] idx = Dependence.Util.sortedIndex(adapter, data, len);
     // Note: As described in Algorithm 1 of reference paper.
-    MWPRanking I = new MWPRanking(idx);
+    int[] idx = MathUtil.sequence(0, len);
+    IntegerArrayQuickSort.sort(idx, (x, y) -> Double.compare(adapter.getDouble(data, x), adapter.getDouble(data, y)));
+    MWPRanking I = new MWPRanking(idx, len);
     long correction = 0;
     for(int j = 0; j < len;) {
-      final double v = adapter.getDouble(data, idx[j]);
```

Comment:
This construct was inserted for efficiency.

Reply:
Done

Reply:
I actually don't know why my version did not incorporate this change (as well as the `final double Z = std > 0 ? Math.abs((U - (0.5 * n1 * n2)) / std) : 0.;`) as I fetched and merged the upstream version